src/main/java/com/ietswise/service/impl/GoogleEventsServiceImpl.java

```diff
@@ -98,4 +109,148 @@ private ZonedDateTime extractDate(JSONObject dateTime) {
         }
         return null;
     }
+
+    /**
+     * Method for getting all free and busy hours of all days of the month
+     * @param tutorId tutor's email
+     * @param year year
+     * @param month month number
+     * @return List<FreeAndBusyHoursOfTheDay>
+     */
+    @Override
+    public List<FreeAndBusyHoursOfTheDay> getEventsByYearAndMonth(String tutorId, int year, int month) {
+        try {
+            ZonedDateTime startOfMonth = YearMonth.of(year, month).atDay(1).atStartOfDay(ZoneId.systemDefault());
+            ZonedDateTime endOfMonth = YearMonth.of(year, month).atEndOfMonth().atStartOfDay(ZoneId.systemDefault());
+
+            URL obj = new URL(createUrl(tutorId, startOfMonth, endOfMonth));
+
+            HttpURLConnection con = (HttpURLConnection) obj.openConnection();
```

Comment:
128-138 Lines of duplicate code. Place them in a separate method along with the code on lines 62-74